components/ShoppingItem/ShoppingItem.js

```diff
@@ -2,13 +2,26 @@ import styled from "styled-components";
 import Link from "next/link";
 import { useState } from "react";
 
-export default function ShoppingItem({ onDeleteItem, shoppingItem }) {
+export default function ShoppingItem({
+  id,
+  onDeleteItem,
+  shoppingItem,
+  onEditItem,
+  onChangeMode,
+  mode,
+}) {
   const [isToBeDeleted, setIsToBeDeleted] = useState(false);
 
+  const [isToBeEdited, setIsToBeEdited] = useState(false);
+
   function toggleIsToBeDeleted() {
     setIsToBeDeleted(!isToBeDeleted);
   }
 
+  function toggleIsToBeEdited() {
+    setIsToBeEdited(!isToBeEdited);
```

Comment:
consider removing this state as it might not be necessary:

export default function ShoppingItem({
  id,
  shoppingItem,
  onDeleteItem,
  onEditItem,
  onChangeMode,
}) {
  const [isBeingDeleted, setIsBeingDeleted] = useState(false);

  function toggleDelete() {
    setIsBeingDeleted(!isBeingDeleted);
  }

  return (
    <ListItem key={shoppingItem.id}>
      {!isBeingDeleted ? (
        <>
          {shoppingItem.quantity} {shoppingItem.name}
          
          <ActionButton
            onClick={() => {
              onChangeMode("edit");
              onEditItem(shoppingItem);
            }}
          >
            Edit
          </ActionButton>
          
          <ActionButton onClick={toggleDelete}>
            Delete
          </ActionButton>
          
          <StyledLink href={`/shoppingItems/${shoppingItem.id}`}>
            Details
          </StyledLink>
        </>
      ) : (
        <DeletionConfirmation>
          Are you sure you want to delete this item?
          <ActionButton onClick={() => onDeleteItem(shoppingItem.id)}>
            Confirm
          </ActionButton>
          <ActionButton onClick={toggleDelete}>
            Cancel
          </ActionButton>
        </DeletionConfirmation>
      )}
    </ListItem>
  );
}

Reply:
Great Johann. Thank you! :)